src/text.rs

```diff
@@ -528,6 +528,33 @@ pub mod unicode {
 mod tests {
     use crate::prelude::*;
 
+    trait ParserTest<'a> {
+        fn test_ok(self: Self, input: &'a str);
+        fn test_err(self: Self, input: &'a str);
+    }
+
+    impl<'a, P: Parser<'a, &'a str, &'a str>> ParserTest<'a> for P {
+        fn test_ok(self: Self, input: &'a str) {
+            assert_eq!(
+                self.parse(input),
+                ParseResult {
+                    output: Some(input),
+                    errs: vec![]
+                }
+            );
+        }
+
+        fn test_err(self: Self, input: &'a str) {
+            assert_eq!(
+                self.parse(input),
+                ParseResult {
+                    output: None,
+                    errs: vec![EmptyErr::default()]
+                }
+            );
+        }
+    }
```

Comment:
Let me know if:
* This trait seems like overkill and you'd rather make these free functions.
* This trait seems generally useful and you'd rather move them to a `cfg(test)` block in `util.rs`.

Reply:
Hmm, I do think this is overkill. Free functions would be a little simpler.

Reply:
Done.